src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -65,6 +66,82 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void ShowLicenseAgreements::operator()(Execution::Context& context) const
+    {
+        const auto& manifest = context.Get<Execution::Data::Manifest>();
+        auto agreements = manifest.CurrentLocalization.Get<AppInstaller::Manifest::Localization::Agreements>();
+
+        if (agreements.empty())
+        {
+            // Nothing to do
+            return;
+        }
+
+        context << Workflow::ShowManifestGlobalInfo;
+        context.Reporter.Info() << std::endl;
+
+        if (m_ensureAcceptance)
+        {
+            context << Workflow::EnsureLicenseAcceptance(/* showPrompt */ true);
+        }
+    }
+
+    void EnsureLicenseAcceptance::operator()(Execution::Context& context) const
+    {
+        if (context.Args.Contains(Execution::Args::Type::AcceptPackageAgreements))
+        {
+            AICLI_LOG(CLI, Info, << "License agreements accepted by CLI flag");
+            return;
+        }
+
+        if (m_showPrompt)
+        {
+            bool accepted = context.Reporter.PromptForBoolResponse(Resource::String::LicenseAgreementPrompt);
+            if (accepted)
+            {
+                AICLI_LOG(CLI, Info, << "License agreements accepted in prompt");
+                return;
+            }
+            else
+            {
+                AICLI_LOG(CLI, Info, << "License agreements not accepted in prompt");
+            }
+        }
+
+        AICLI_LOG(CLI, Error, << "License not agreed to.");
+        context.Reporter.Error() << Resource::String::LicenseNotAgreedTo << std::endl;
+        AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_LICENSE_NOT_ACCEPTED);
+    }
+
+    void EnsureLicenseAcceptanceForMultipleInstallers(Execution::Context& context)
+    {
+        bool hasLicenseAgreements = false;
+        for (auto package : context.Get<Execution::Data::PackagesToInstall>())
+        {
+            // Show agreements for each package in a sub-context
+            auto showContextPtr = context.Clone();
+            Execution::Context& showContext = *showContextPtr;
+
+            showContext.Add<Execution::Data::Manifest>(package.Manifest);
+
+            showContext <<
+                Workflow::ReportManifestIdentity <<
+                Workflow::ShowLicenseAgreements(/* ensureAcceptance */ false);
+            if (showContext.IsTerminated())
+            {
+                AICLI_TERMINATE_CONTEXT(showContext.GetTerminationHR());
+            }
+
+            hasLicenseAgreements |= !package.Manifest.CurrentLocalization.Get<AppInstaller::Manifest::Localization::Agreements>().empty();
+        }
+
+        // If any package has agreements, ensure they are accepted
+        if (hasLicenseAgreements)
+        {
+            context << Workflow::EnsureLicenseAcceptance(/* showPrompt */ false);

```

Comment:
>false [](http://example.com/codeflow?start=74&length=5)

Do you intend to only allow --aceept-agreements in import case?

Reply:
Yes.